src/test/signalgentests.py

```diff
@@ -0,0 +1,83 @@
+import unittest
+from testhelper import TestHelp
+TestHelp().chwd()
+from semanticanalyzer import Semantic
+from lark import Tree
+from lark import lexer
+
+Token = lexer.Token
+
+class TestSignalGen(unittest.TestCase):
+
+    def setUp(self):
+        self.help = TestHelp()
+        self.semantic = Semantic(Tree('start', []))
+
+
+    def test_measureSigGen(self):
+        input_measure = Tree('measure', [])
+        expected = [{'type':'measure'}]
+        signal = self.semantic.measure_to_signal(input_measure)
+        self.assertEqual(expected, signal, "Measure signal not valid")
+
```

Comment:
Can you make another unit test here?

Copy the test_measureSignalGen, and make the tree also create a dynamic signal.
Then assert that the two lists aren't equal